### PR TITLE
Fix unsupported characters

### LINKED ISSUE #680: Block offline mode for files that contain characters prohibited by linux/android
android.system.ErrnoException: open failed: EPERM (Operation not permitted)
        at libcore.io.Linux.open(Linux.java)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:567)
        at libcore.io.BlockGuardOs.open(BlockGuardOs.java:273)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:567)
        at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7728)
        at libcore.io.IoBridge.open(IoBridge.java:561)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:236)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:186)
        at kotlin.io.FilesKt__UtilsKt.copyTo(Utils.kt:236)
        at kotlin.io.FilesKt__UtilsKt.copyTo$default(Utils.kt:217)
        at com.infomaniak.drive.utils.Utils.moveCacheFileToOffline(Utils.kt:254)
        at com.infomaniak.drive.views.FileInfoActionsView.downloadAsOfflineFile(FileInfoActionsView.kt:217)
        at com.infomaniak.drive.views.FileInfoActionsView$OnItemClickListener$DefaultImpls.availableOfflineSwitched(FileInfoActionsView.kt:441)

Exemple: `<` or `>`

https://sentry.infomaniak.com/organizations/infomaniak/issues/9793/?project=3

### PR DESCRIPTION
**Pull request by FabianDevel**
closes #680 

Do not start offline downloading if file name contains characters forbidden by android/linux filesystem
Display a snack bar for the user on single action and notifications for multiselect actions
Close the bottomSheet after switching offline/online to be consistent with other functionnalities

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/utils/Utils.kt [hunk 1]
```diff
@@ -274,6 +285,7 @@ object Utils {
 
         workManager
             .enqueueUniqueWork(DownloadWorker.TAG, ExistingWorkPolicy.APPEND_OR_REPLACE, downloadRequest)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/drive/utils/Utils.kt:287]:
```suggestion
        workManager.enqueueUniqueWork(DownloadWorker.TAG, ExistingWorkPolicy.APPEND_OR_REPLACE, downloadRequest)
```

File: app/src/main/res/values-de/strings.xml [hunk 2]
```diff
@@ -433,6 +439,10 @@
     <string name="fileShareLinkSettingsTitle">Erweiterte Einstellungen des Freigabelinks</string>
     <string name="fileTypeFilterTitle">Dateityp</string>
     <string name="filtersTitle">Filter</string>
+    <plurals name="folder">
+        <item quantity="one">1 Ordner</item>
+        <item quantity="other">%d Ordner</item>
+    </plurals>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-de/strings.xml:445]:
```suggestion
```

File: app/src/main/res/values-de/strings.xml [hunk 3]
```diff
@@ -266,6 +268,10 @@
     <string name="errorShareLink">Fehler bei der Erstellung des Links</string>
     <string name="favoritesNoFile">Derzeit keine Favoriten</string>
     <string name="favoritesTitle">Favoriten</string>
+    <plurals name="file">
+        <item quantity="one">1 Datei</item>
+        <item quantity="other">%d Dateien</item>
+    </plurals>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-de/strings.xml:274]:
```suggestion
```

File: app/src/main/res/values-de/strings.xml [hunk 4]
```diff
@@ -235,6 +236,7 @@
     <string name="dropboxSharedLinkTitle">Dropbox-Freigabelink</string>
     <string name="editCategoryInfoDescription">Die Änderung ist für alle Benutzer des kDrive sichtbar, die Zugriff auf die Kategorien besitzen.</string>
     <string name="editCategoryTitle">Kategorie bearbeiten</string>
+    <string name="emptyFolder">Leerer Ordner</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-de/strings.xml:239]:
```suggestion
```

File: app/src/main/res/values-de/strings.xml [hunk 5]
```diff
@@ -182,6 +182,7 @@
     <string name="cloudStorageQuerySearchImpossibleSelectDrive">Suche nicht möglich, es wurde kein Drive ausgewählt</string>
     <string name="commonFolderDescription">Teilen Sie diesen Ordner mit den Benutzern des kDrive %s</string>
     <string name="commonFolderTitle">Gemeinsamer Ordner</string>
+    <string name="content">Inhalt</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-de/strings.xml:185]:
```suggestion
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]
Update strings in all 5 files like I did in the comments for the german one.

File: app/src/main/res/layout/view_file_info_actions.xml [hunk 6]
```diff
@@ -15,742 +15,750 @@
   ~ You should have received a copy of the GNU General Public License
   ~ along with this program.  If not, see <http://www.gnu.org/licenses/>.
   -->
-<androidx.core.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+
```
Comment (KevinBoulongne) [inline app/src/main/res/layout/view_file_info_actions.xml:18]:
```suggestion
```

File: app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt [hunk 7]
```diff
@@ -387,8 +387,11 @@ abstract class MultiSelectFragment(private val matomoCategory: String) : Fragmen
                 }
             }
         } else {
-            Utils.downloadAsOfflineFile(requireContext(), file)
-            file.isOffline = true
+            if (!Utils.downloadAsOfflineFile(requireContext(), file)) {
+                requireActivity().showSnackbar(requireContext().getString(R.string.snackBarInvalidFileNameError, file.name))
+            } else {
+                file.isOffline = true
+            }
```
[the remark this answers is not in the document]
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/multiSelect/MultiSelectFragment.kt:394]: done

File: app/src/main/java/com/infomaniak/drive/utils/Utils.kt (hunk 1 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/utils/Utils.kt:287]: done

File: app/src/main/res/layout/view_file_info_actions.xml (hunk 6 above)
Comment (FabianDevel) [reply app/src/main/res/layout/view_file_info_actions.xml]: done

File: app/src/main/res/values-de/strings.xml (hunk 2 above)
Comment (FabianDevel) [reply app/src/main/res/values-de/strings.xml]: done

File: app/src/main/java/com/infomaniak/drive/utils/Utils.kt (hunk 1 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/utils/Utils.kt]: done

**Review by KevinBoulongne — APPROVED** [review]

**Review by JorisBodin — CHANGES_REQUESTED** [review]
We need a clearer message and now we can't see the progression in the filelist

**Review by JorisBodin — APPROVED** [review]

**Pull request merged**